container_restore_opts.go

```diff
@@ -148,3 +151,35 @@ func WithRestoreRW(ctx context.Context, id string, client *Client, checkpoint Im
 		return nil
 	}
 }
+
+// Check if the Spec requires a terminal to be allocated  -> { terminal: true or false }
+func CheckSpecTTY(ctx context.Context, client *Client, desc ocispec.Descriptor) (*bool, error) {
```

Comment:
While the intention of your change as a whole was to use this to check if a restored container needs a tty, this file is solely for Restore related opts function as of now (`WithFooBar() NewContainerOpts`). Although maybe there's another way to get the info you're after without this altogether